gax/src/streamingCalls/streaming.ts

```diff
@@ -522,4 +363,232 @@ export class StreamProxy extends duplexify implements GRPCCallResult {
+        if (
+          typeof maxRetries !== undefined ||
+          typeof maxRetries !== undefined
+        ) {
+          if (this.shouldRetryRequest(error, retry)) {
+            if (maxRetries && totalTimeout) {
+              const newError = new GoogleError(
+                'Cannot set both totalTimeoutMillis and maxRetries ' +
+                  'in backoffSettings.'
+              );
+              newError.code = Status.INVALID_ARGUMENT;
+              // clean up the request stream and retryStreams, silently destroy it on the request stream
+              // but do raise it on destructin of the retryStream so the consumer can see it
+              requestStream.destroy();
+              retryStream.destroy(newError);
+
+              return retryStream;
+            } else {
+              // check for exceeding timeout or max retries
+
+              try {
+                this.throwIfMaxRetriesOrTotalTimeoutExceeded(
+                  deadline,
+                  maxRetries!,
+                  timeout!,
+                  error,
+                  totalTimeout,
+                  retries
+                );
+              } catch (error: unknown) {
+                const e = GoogleError.parseGRPCStatusDetails(
+                  error as GoogleError
+                );
+                // clean up the request stream and retryStreams, silently destroy it on the request stream
+                // but do raise it on destruction of the retryStream so the consumer can see it
+                requestStream.destroy();
+                retryStream.destroy(e);
+
+                return retryStream;
+              }
+
+              const delayMult = retry.backoffSettings.retryDelayMultiplier;
+              const maxDelay = retry.backoffSettings.maxRetryDelayMillis;
+              const timeoutMult = retry.backoffSettings.rpcTimeoutMultiplier;
+              const maxTimeout = retry.backoffSettings.maxRpcTimeoutMillis;
+              let delay = retry.backoffSettings.initialRetryDelayMillis;
+              // calculate new deadlines
+              const toSleep = Math.random() * delay;
+              const calculateTimeoutAndResumptionFunction = () => {
+                setTimeout(() => {
+                  // only do timeout calculations if not using maxRetries
+                  if (timeout) {
+                    now = new Date();
+                    delay = Math.min(delay * delayMult, maxDelay);
+                    const timeoutCal =
+                      timeout && timeoutMult ? timeout * timeoutMult : 0;
+                    const rpcTimeout = maxTimeout ? maxTimeout : 0;
+                    const newDeadline = deadline ? deadline - now.getTime() : 0;
+                    timeout = Math.min(timeoutCal, rpcTimeout, newDeadline);
+                  }
+
+                  retries++;
+                  let retryArgument = this.argument! as unknown as RequestType;
```

Comment:
usually converting to unknown to type is considered unsafe. Do we know for sure it will be RequestType? If so we can just coerce it to RequestType? Maybe TS is concerned that argument won't exist, in which case you could wrap it in an if-branch and then perhaps it won't complain about coercing the type (not sure).

Reply:
This is a good call. I can't remember why I went to unknown then requestType, but coercing to RequestType does seem to be viable. I believe we do know it will be ReuqestType based on comments [like this](https://github.com/googleapis/gax-nodejs/blob/6e040f665fe750bd7ace98bdd2d50537e559fcd0/gax/src/apitypes.ts#L30-L35) and some other stricter typing throughout gax. TS is concerned it won't exist because it is legal to be undefined. I will mess around with a combo of coercing and if statements to address this and the other comment